bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -611,14 +612,14 @@ private IUnknown createControllerInitializationCallback(int previousAttempts) {
 			SWT.error(SWT.ERROR_INVALID_ARGUMENT, null,
 					" Edge instance with same data folder but different environment options already exists");
 		}
-		switch ((int) result) {
+		switch (result) {
```

Comment:
if it is a long, maybe just use a switch here and instead if/else chain would be more future proof?

Reply:
The issue is that we use integers to represent HRESULT values in SWT all over the place. This also includes the "improper" signing of these values, as in the Windows API they have always been unsigned values (which Java is not able to represent). That's why we also have constants (like `E_ABORT` used in this switch) represented as signed integer values. 

For that reason, I would propose to stick to integer in all the result code processing for now. As a separate topic, I would be in favor of completely replacing all tretments of HRESULT as integer in the OS APIs and constants with long values.

Reply:
> This also includes the "improper" signing of these values, as in the Windows API they have always been unsigned values (which Java is not able to represent).

I don't think this is particular true. Java can represent an "unsigned" long but one has to be careful when comparing them or perform math, beside from that you can do whatever you want with them (including equals comparison).

Also I'm not sure that "because we do it everywhere" is a good argument here ... maybe this is a good reason to start with something new, e.g. make a `HRESULT(long)` record and use that whenever possible with a `HRESULT#match(...)` that makes sure one do no "illegal" operations on this here (like using it as a long).

Reply:
As said: I agree that there is some need for general improvement and I will create an issue for that. But starting with some locally introduced improvement for HRESULT processing that is incompatible with the OS APIs (that currently treat HRESULT as int) does not make sense to me. The current assumption is that HRESULT is an int, so we need to treat it like that. If we want to replace that (of which I would be in favor), we need to do that everywhere.